src/Microsoft.OData.Client/Metadata/ODataTypeInfo.cs

```diff
@@ -396,7 +396,7 @@ private static KeyKind IsKeyProperty(PropertyInfo propertyInfo, KeyAttribute dat
                 order = newOrder;
                 keyKind = newKind;
             }
-            else if (propertyName.EndsWith("ID", StringComparison.Ordinal))
+            else if (propertyName.EndsWith("ID", StringComparison.OrdinalIgnoreCase))

```

Comment:
Could we match this instead to match ending with Id or ID since this seems to be mostly the convention users use. This would eliminate matching other words which end with id in English or other alphabets which if needed the user can use the Key attribute.
So this matches UserId, UserID, and wont match words such as Avoid.
```suggestion
            else if (propertyName.EndsWith("ID", StringComparison.Ordinal) || propertyName.EndsWith("Id", StringComparison.Ordinal))

```

Reply:
Actually, I think it should match what is in the ConventionBuilder:
https://github.com/OData/ModelBuilder/blob/main/src/Microsoft.OData.ModelBuilder/Conventions/EntityKeyConvention.cs#L51
```
(p.Name.Equals(entityType.Name + "Id", StringComparison.OrdinalIgnoreCase) || p.Name.Equals("Id", StringComparison.OrdinalIgnoreCase)
```
This way it checks for Id properties, or properties that are the EntityType.Name + "Id"  which avoids the stray word ending in id.

Reply:
@marabooy any thoughts on this?

Reply:
This is better thanls